staff/views/index.py

```diff
@@ -0,0 +1,80 @@
+import functools
+import itertools
+import operator
+
+from django.db.models import Q
+from django.template.response import TemplateResponse
+from django.utils.decorators import method_decorator
+from django.views.generic import View
+
+from jobserver.authorization import CoreDeveloper
+from jobserver.authorization.decorators import require_role
+from jobserver.models import Backend, User
+
+
+# configure searchable models here, each must have get_staff_url defined
+configured_searches = [
+    {"model": Backend, "fields": ["name", "slug"], "order_by": "name"},
+    {
+        "model": User,
+        "fields": ["first_name", "last_name", "username"],
+        "order_by": "username",
+    },
+]
+
+
+def get_results(q):
+    """
+    Turn a search term into queries for each configured model
+
+    This takes the given search term and for each model configured to be
+    searchable does:
+
+        1. creates a Q(field=search_term) instance
+        2. combines them with a bitwise OR, typically expressed with a | (pipe)
```

Comment:
nit: it's not technically a bitwise OR, but a logical OR (although it uses the syntax for a bitwise operation).

Reply:
ooh thank you, I almost certainly just read the `|` docs!